lib/arborist/build-ideal-tree.js

```diff
@@ -326,18 +337,11 @@ module.exports = cls => class IdealTreeBuilder extends cls {
     return Promise.all(add.map(s => {
```

Comment:
how come this function isn’t `async`, since it returns a Promise? It’d let you use await here.

Reply:
I just went with the minimal changes here, I just refactored that first part of the method into `_retrieveSpecName` since it was getting really confusing to follow, but otherwise I'm not biting the bullet to a full `async` refactor, specially in the context of a fix-PR 😊